sql/moz-fx-data-shared-prod/mozilla_vpn_derived/fxa_attribution_v1/query.sql

```diff
@@ -51,18 +51,41 @@ flows AS (
     fxa_content_auth_stdout_events
   WHERE
     IF(@date IS NULL, partition_date < CURRENT_DATE, partition_date = @date)
-    -- cannot filter service because
-    AND (
+    AND flow_id IS NOT NULL
+  GROUP BY
+    flow_id
+  HAVING
+    LOGICAL_OR(
       service = "guardian-vpn"
-      -- service is missing for these event types
+      OR entrypoint = "www.mozilla.org-vpn-product-page"
+      OR utm_source = "www.mozilla.org-vpn-product-page"
+      OR (
+        event_type = "fxa_rp_button - view"
+        AND (
+          -- The www.mozilla.org navbar CTA button was changed to link to VPN for Firefox users
+          -- on 2021-09-02, then attribution was implemented for it on 2021-09-15.
+          (
+            service IS NULL
+            AND utm_source = "www.mozilla.org"
+            AND utm_campaign = "navigation"
+            AND (partition_date BETWEEN "2021-09-15" AND "2021-12-08")
```

Comment:
Previously all `fxa_rp_button - view` events with null `service` were being included, but the the vast majority of such events are from the "Join Firefox" button on the Firefox homepage and various links on the Firefox "What's New" pages, so this is significantly more targeted to the VPN CTA button.